### ISSUE TITLE
Wrongly reports missing copyright/licensing information for a Qt RCC file

### ISSUE THREAD

**Issue by ingokloecker** [fsfe/reuse-tool#895]
This is a regression compared to version 2.1.

For the file `resources.qrc` with content
```
<!--
 SPDX-FileCopyrightText: 2020 Dimitris Kardarakos <dimkard@posteo.net>
 SPDX-License-Identifier: BSD-2-Clause
-->
<RCC>
    <qresource prefix="/">
        <file alias="Main.qml">contents/ui/Main.qml</file>
        <file alias="GlobalDrawer.qml">contents/ui/GlobalDrawer.qml</file>
        <file alias="ScheduleView.qml">contents/ui/ScheduleView.qml</file>
        <file alias="FavoritesView.qml">contents/ui/FavoritesView.qml</file>
        <file alias="MapView.qml">contents/ui/MapView.qml</file>
        <file alias="Conferences.qml">contents/ui/Conferences.qml</file>
        <file alias="EventInfo.qml">contents/ui/EventInfo.qml</file>
        <file alias="SettingsView.qml">contents/ui/SettingsView.qml</file>
    </qresource>
</RCC>
```
the tool reports
```
# MISSING COPYRIGHT AND LICENSING INFORMATION

The following files have no copyright and licensing information:
* src/resources.qrc


# SUMMARY

* Bad licenses: 0
* Deprecated licenses: 0
* Licenses without file extension: 0
* Missing licenses: 0
* Unused licenses: 0
* Used licenses: BSD-2-Clause, CC0-1.0, GPL-3.0-or-later, LGPL-3.0-or-later, CC-BY-SA-4.0, BSD-3-Clause
* Read errors: 0
* files with copyright information: 79 / 80
* files with license information: 79 / 80

Unfortunately, your project is not compliant with version 3.0 of the REUSE Specification :-(
```

I have bisected this. The bad commit is https://git.fsfe.org/reuse/tool/commit/ead44eac0df7a3c8de695aef88b459b93296ae4b. Apparently, this change makes the reuse tool treat the above file as "uncommentable" which is clearly wrong.

Digging further it looks like
```
".qrc": UncommentableCommentStyle
```
in src/reuse/comment.py is why reuse treats the above file as "uncommentable".

This was added by https://git.fsfe.org/reuse/tool/commit/5895f3682ba09ced994fd3be714ab4b4598e4e58 with comment
> .qrc: Those are Qt XML files containing indexes to resource files.
      Added as Uncommentable because Qt Creator IDE rewrites the file
      when saving, loosing copyrights each time therefore loosing history.

I'm not convinced that the behavior of one IDE is a good reason for treating .qrc files as "uncommentable". In particular, it gives a wrong result (claims non-compliance with REUSE 3.0). The above file is clearly compliant with version 3.0 of the REUSE Specification (which version 2.1 of the tool correctly reports).

I propose to revert https://git.fsfe.org/reuse/tool/commit/5895f3682ba09ced994fd3be714ab4b4598e4e58. People using Qt Creator will have to provide the copyright information for their .qrc (and other Qt XML files) externally while we can continue adding the copyright information directly in the .qrc files.

**Comment by fmuellner** [comment]
> This was added by https://git.fsfe.org/reuse/tool/commit/5895f3682ba09ced994fd3be714ab4b4598e4e58

That commit also broke XML UI definitions for GTK projects, because they usually use the same `.ui` suffix as Qt Creator.

**Comment by carmenbianca** [comment]
You're right this is a bug. Thanks for flagging this. There are two things simultaneously not quite right:

- The files marked uncommentable are, in fact, commentable. Marking them as uncommentable _was_ fine at the time of #755: it just meant that when doing `reuse annotate`, a `.license` file would be created instead of either failing or adding a comment.
- However, in #825, the semantics of what it means to be an uncommentable file slightly changed. Now, uncommentable files are not parsed at all for REUSE info.

I will get to work on writing a quick patch to right this situation, and getting a release out the door soon (today?). Most likely I will do this:

1. Files that are in fact commentable shouldn't be marked commentable, even if wisdom dictates that you shouldn't put licensing information inside of them.
2. I will revert #825. That feature is likely to cause grief in the future as well.

Thanks!

**Comment by biljkus** [comment]
We also have an issue with **.csv**. One day ago the pipeline with the REUSE job passed correctly, and today it complains that there is no copyright/license when there is: https://gitlab.com/cttc-lena/nr/-/pipelines/1143245343.

**Comment by ingokloecker** [comment]
Thanks a lot for the swift reaction!

**Comment by carmenbianca** [comment]
Fixed in v3.0.1, hopefully. Let me know if something is still awry.

**Comment by biljkus** [comment]
Thanks for the quick fix. It works for our issue with .csv.
